core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -1068,8 +1067,7 @@ class KafkaServer(
         if (socketServer != null)
           CoreUtils.swallow(socketServer.shutdown(), this)
         unregisterCurrentControllerIdMetric()
-        if (metrics != null)
-          CoreUtils.swallow(metrics.close(), this)
+        Utils.closeQuietly(metrics, "metrics")
         if (brokerTopicStats != null)
           CoreUtils.swallow(brokerTopicStats.close(), this)
```

Comment:
Could you please make `brokerTopicStats` extend `Closeable`?

Reply:
Sounds great! Add it in 21f9b18.